source/common/router/router.h

```diff
@@ -231,10 +235,12 @@ class FilterConfig {
   const bool start_child_span_;
   const bool suppress_envoy_headers_;
   const bool respect_expected_rq_timeout_;
+  const bool add_request_date_header_;
   // TODO(xyu-stripe): Make this a bitset to keep cluster memory footprint down.
   HeaderVectorPtr strict_check_headers_;
   std::list<AccessLog::InstanceSharedPtr> upstream_logs_;
   Http::Context& http_context_;
+  Http::SlowDateProviderImpl date_provider_;
```

Comment:
I'm interested in perf.... did you consider the cached date provider to avoid creating a new date string every time?

Reply:
I had considered using the `TlsCachingDateProviderImpl`, but realized that it only updates every 500ms. Since this header is being sent with outbound requests (as opposed to responses), I figured it was more important to make sure this header is accurate. It's also opt-in, so no existing consumers will be affected negatively (that said, I'm not sure what the actual cost of `SlowDateProviderImpl`'s processing is)

Reply:
Closing this out, it makes sense to monitor and change if you notice issues as it's opt-in